bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/toml/Utils.java

```diff
@@ -66,6 +93,65 @@ static Object getTomlTypeString(TomlNode tomlNode) {
         }
     }
 
+    static Object getBalValue(TomlNode tomlNode, Set<TomlNode> visitedNodes) {
+        visitedNodes.add(tomlNode);
+        switch (tomlNode.kind()) {
+            case STRING:
+                return StringUtils.fromString(((TomlStringValueNode) tomlNode).getValue());
+            case INTEGER:
+                return ((TomlLongValueNode) tomlNode).getValue();
+            case DOUBLE:
+                return ((TomlDoubleValueNodeNode) tomlNode).getValue();
+            case BOOLEAN:
+                return ((TomlBooleanValueNode) tomlNode).getValue();
+            case KEY_VALUE:
+                return getBalValue(((TomlKeyValueNode) tomlNode).value(), visitedNodes);
+            case ARRAY:
+                TomlArrayValueNode arrayValueNode = (TomlArrayValueNode) tomlNode;
+                ListInitialValueEntry[] arrayValues = new ListInitialValueEntry[arrayValueNode.elements().size()];
+                List<TomlValueNode> elements = arrayValueNode.elements();
+                int count = 0;
+                for (TomlValueNode tomlValueNode : elements) {
+                    arrayValues[count++] = new ListInitialValueEntry.ExpressionEntry(getBalValue(tomlValueNode,
+                                                                                                 visitedNodes));
+                }
+                return new ArrayValueImpl(TypeCreator.createArrayType(TYPE_READONLY_ANYDATA_INTERSECTION, true),
+                                          arrayValues.length, arrayValues);
```

Comment:
If there are multiple lines under a case label, shall we create separate methods out of them? This is because it is recommended to have only single line case statements to reduce code complexity

Reply:
Refactored